### PR TITLE
Solarescarlos101/cfd 153 create api endpoint to fetch subactivities for events

### PR DESCRIPTION
**Pull request by solarescarlos**
### Description

Created the getSubactivities endpoint in packages/api/src/router/activity.ts as described in https://linear.app/uoftblueprint/issue/CFD-153/create-api-endpoint-to-fetch-subactivities-for-events.
When given an activity id, it returns all subactivities in that activity. Included some console logs.

### Reason for Change

https://linear.app/uoftblueprint/issue/CFD-153/create-api-endpoint-to-fetch-subactivities-for-events

Completes CFD ticket https://linear.app/uoftblueprint/issue/CFD-153/create-api-endpoint-to-fetch-subactivities-for-events

### Type of change

- [X] New feature (non-breaking change which adds functionality)

### Testing

- First I added some console log statements within the endpoint. Then, I ran pnpm dev and created a new activity with its own subactivity. I then navigated to the tRPC panel (http://localhost:3000/api/panel) and called getSubactivities under the activity router. I used id 100 (use the id of the activity you created).
Screenshot (115)

### CODE REVIEW

**Comment by solarescarlos** [comment]
Currently fixing format

File: packages/api/src/router/activity.ts [hunk 1]
```diff
@@ -76,4 +81,22 @@ export const activityRouter = createTRPCRouter({
         },
       });
     }),
+  getSubactivities: publicProcedure
+    .input(z.object({ id: z.number() })) // I assumed "event" and "activity" are used interchangeably in the ticket
+    .query(async ({ ctx, input }) => {
+      console.log("Received input:", input);
```
Comment (sarinali) [inline packages/api/src/router/activity.ts:87]: No need for this logging statement can you delete this?

**Review by sarinali — CHANGES_REQUESTED** [review]

File: packages/api/src/router/activity.ts [hunk 2]
```diff
@@ -76,4 +81,20 @@ export const activityRouter = createTRPCRouter({
...
+      const result = await ctx.db.activity.findUnique({
+        where: {
+          id: input.id,
+        },
+        select: {
+          subactivities: true,
+        },
+      });
+
+      console.log("Query result:", result);
```
Comment (sarinali) [inline packages/api/src/router/activity.ts:96]: remove this one as well

**Review by sarinali — CHANGES_REQUESTED** [review]

**Review by sarinali — APPROVED** [review]
LGTM

File: packages/api/src/router/activity.ts [hunk 3]
```diff
@@ -76,4 +81,18 @@ export const activityRouter = createTRPCRouter({
         },
       });
     }),
+  getSubactivities: publicProcedure
```
Comment (sarinali) [inline packages/api/src/router/activity.ts:84]: should be protectedprocedure actually

**Review by sarinali — CHANGES_REQUESTED** [review]

**Review by sarinali — APPROVED** [review]
LGTM

**Pull request merged**